sqlmesh/dbt/model.py

```diff
@@ -156,23 +156,56 @@ def model_kind(self, target: TargetConfig) -> ModelKind:
             return ModelKind(name=ModelKindName.EMBEDDED)
         raise ConfigError(f"{materialization.value} materialization not supported.")
 
-    @property
-    def sql_no_config(self) -> str:
-        matches = re.findall(r"{{\s*config\(", self.sql)
-        if matches:
-            config_macro_start = self.sql.index(matches[0])
-            cursor = config_macro_start
+    def _sql_extract_config(self) -> t.Tuple[SqlStr, SqlStr]:
+        def match_end(str: str, start: int) -> int:
```

Comment:
Observe for example what's going on here:

```python
>>> str = 5
>>> str(5)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'int' object is not callable
```

```python
>>> str(5)  # On a new REPL context where `str` hasn't been set a value
'5'
```

Reply:
Right, that's a constructor.

Reply:
What do you suggest instead?

Reply:
Maybe this won't cause any problems, just thought it'd be worth discussing because the semantics seemed non-obvious to me.

> What do you suggest instead?

What about `string: str`?

Reply:
When choosing a name for your variables, choose descriptive names that reflect the purpose and content of the variable. Avoid using names that are too generic or easily confused with Python's built-in functions or keywords. For example, don't use names like "list", "str", "int", or "tuple" for your variables.

> If a function argument’s name clashes with a reserved keyword, it is generally better to append a single trailing underscore rather than use an abbreviation or spelling corruption. Thus class_ is better than clss. (Perhaps better is to avoid such clashes by using a synonym.)

https://peps.python.org/pep-0008/#function-and-method-arguments